### PR TITLE
Add gen info for signals

### PR DESCRIPTION
**Pull request by OzAmram**
Add new flag to H5 maker to save generator level info for signals (needed for Lund reweighting)
There is a custom function needed for each signal to correct parse the gen level info and save what is needed. Updated the README with an explanation

### CODE REVIEW

File: H5_maker/README.md [hunk 1]
```diff
@@ -78,6 +78,15 @@ variations.
 
 Variations of the **preselection\_eff** for JES and JER systematic variations are also saved (eg `preselection_eff_JES_up`)
 
+**gen_info** (N_Prongs, 4) floags :
```
Comment (lmoureaux) [inline H5_maker/README.md]:
```suggestion
**gen_info** (N_Prongs, 4) floats : 
```

File: H5_maker/README.md [hunk 2]
```diff
@@ -78,6 +78,15 @@ variations.
 
 Variations of the **preselection\_eff** for JES and JER systematic variations are also saved (eg `preselection_eff_JES_up`)
 
+**gen_info** (N_Prongs, 4) floags : 
+If the `--gen` flag is used (followed by a label for a specific signal), additional generator level information of the
+event under this column . For signals, generator level information (specifically the 4-vectors of the
```
Comment (lmoureaux) [inline H5_maker/README.md]:
```suggestion
If the `--gen` flag is used (followed by a label for a specific signal), additional generator level information 
is included. For signals, generator level information (specifically the 4-vectors of the
```

**Review by lmoureaux — COMMENTED** [review]
I'm not sure your code finds meaningful kinematics for the quarks. Specifically, since you don't use status 1 particles, you likely select copies coming directly from the matrix element generator (because event contents are usually ordered that way). These copies lack corrections from the parton shower, which can be quite dramatic at high mass. For instance, the resonant particle will have pT = 0 because ISR is missing completely at this stage. The pT added by the parton shower is O(m), so neglecting it would be a quite bold approximation.

One trick can be, after you find the quarks, to walk down the tree and collect all status 1 particles that ultimately descend from each of them. You can then sum the four-vectors of these child particles to recover the "corrected" quark momentum taking all ISR and FSR corrections into account (unfortunately, it's not possible to disentangle them).

Also found two typos in the docs.

**Comment by OzAmram** [comment]
Hmm ok, let me try the method you suggest. 

These generator level particles are mostly needed to assess a matching systematic between the found subjets and the 'true' prongs. So I don't think they have to be perfect, but probably better to do something more proper

Thanks for finding those typos :)

**Comment by OzAmram** [comment]
It seems like it is not always possible to follow the decay chain down because NanoAOD only stores one mother index. So some hadrons that get formed from multiple quarks and only one of them ends up in the history 

Do you have any examples @lmoureaux ? Or I can ask around to find out what people use

**Comment by lmoureaux** [comment]
I did this in Rivet for my thesis (on top of AOD), so the implementation would be irrelevant here. Just thinking now that my requirements were more complicated, so picking the copy of the quark with the right [Pythia status code](https://pythia.org/latest-manual/ParticleProperties.html) might be sufficient for you.

**Comment by OzAmram** [comment]
Ok I think I will stick with what I have then. Based on all the tests that I did for each signal the current selection seems to work quite well for my purposes

**Pull request merged**